agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +204,635 @@ def waitpid(pid):
+                        )
+                    else:
+                        time.sleep(0.1)
+
+    @staticmethod
+    def wait(chan, logger):
+        """wait - Wait for the Tool Data Sink to report back success or
+        failure regarding the Tool Meister environment setup.
+        """
+        status = ""
+        for data in chan.fetch_json(logger):
+            # We expect the payload to look like:
+            #   { "kind": "ds",
+            #     "action": "startup",
+            #     "status": "success|failure"
+            #   }
+            try:
+                kind = data["kind"]
+                action = data["action"]
+                status = data["status"]
+            except KeyError:
+                logger.warning("unrecognized data payload in message, '%r'", data)
+                continue
+            else:
+                if kind != "ds":
+                    logger.warning("unrecognized kind field in message, '%r'", data)
+                    continue
+                if action != "startup":
+                    logger.warning("unrecognized action field in message, '%r'", data)
+                    continue
+                break
+        return 0 if status == "success" else 1
+
+
+class RedisServer(BaseServer):
+    """RedisServer - an encapsulation of the handling of the Redis server
+    specification and methods to optionally create and manage an instance.
+    """
+
+    envi_var = "PBENCH_REDIS_SERVER_CFG"
+    def_port = def_redis_port
+    bad_port_ret_code = ReturnCode.BADREDISPORT
+
+    # Redis server configuration template for pbench's use
+    conf_tmpl = """bind {bind_host_names}
+daemonize yes
+dir {tm_dir}
+dbfilename pbench-redis.rdb
+logfile {tm_dir}/redis.log
+loglevel notice
+pidfile {tm_dir}/redis_{redis_port:d}.pid
+port {redis_port:d}
+"""
+
+    def __init__(self, spec):
+        super().__init__()
+        if spec:
+            parts = spec.split(":", 1)
+            if len(parts) != 2:
+                raise self.Err(
+                    f"Bad Redis server specified, {spec!r}", ReturnCode.BADREDISARG
+                )
+            try:
+                self.port = int(parts[1])
+            except Exception as exc:
+                raise self.Err(
+                    f"Bad Redis port specified, {spec!r}", self.bad_port_ret_code
+                ) from exc
+            else:
+                self.host = parts[0]
+            # This instance is created by somebody else.
+            self._managed = False
+        else:
+            self.port = None
+            self.host = None
+            # This instance will be managed by us.
+            self._managed = True
+        self.local_host = None
+        self.pid_file = None
+
+    def start(self, tm_dir, full_hostname, logger):
+        """start_redis - configure and start a Redis server.
+
+        Returns a tuple of the Redis server host name, the host name to be used to
+        connect to it locally, and its port.
```

Comment:
Not in any obvious way (i.e., I can't find a `return` statement in this function...).